src/api/procedures/utils.ts

```diff
@@ -322,3 +330,166 @@ export function assertRequirementsNotTooComplex(
+export async function assertTransferTickerAuthorizationValid(
+  context: Context,
+  data: GenericAuthorizationData
+): Promise<void> {
+  const reservation = new TickerReservation({ ticker: data.value }, context);
+  const { status } = await reservation.details();
+  if (status === TickerReservationStatus.Free) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'The Ticker is not reserved',
+    });
+  }
+  if (status === TickerReservationStatus.TokenCreated) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'The ticker has already been used to create an Asset',
+    });
+  }
+}
+
+/** asserts valid transfer asset ownership authorization */
+export async function assertTransferAssetOwnershipAuthorizationValid(
+  context: Context,
+  data: GenericAuthorizationData
+): Promise<void> {
+  const token = new SecurityToken({ ticker: data.value }, context);
+  if (!(await token.exists()))
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'The Asset does not exist',
+    });
+}
+
+/**
+ * asserts valid join identity authorization request
+ */
+export async function assertJoinIdentityAuthorizationValid(
+  authRequest: AuthorizationRequest
+): Promise<void> {
+  // https://github.com/PolymathNetwork/Polymesh/blob/5fec16fdfb05dc9022508880503e835ae9c1776c/pallets/identity/src/keys.rs#L470
+  const identity = authRequest.issuer;
+  if (!(await identity.hasValidCdd())) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'Issuing Identity does not have a valid CDD claim',
+    });
+  }
+}
+
+/** asserts valid add relayer paying key authorization */
+export async function assertAddRelayerPayingKeyAuthorizationValid(
+  data: AddRelayerPayingKeyAuthorizationData
+): Promise<void> {
+  const subsidy = data.value;
+  const beneficiaryIdentity = await subsidy.beneficiary.getIdentity();
+  if (!beneficiaryIdentity) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'Beneficiary Account does not have an Identity',
+    });
+  }
+  if (!(await beneficiaryIdentity.hasValidCdd())) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'Beneficiary Account does not have a valid CDD Claim',
+    });
+  }
+
+  const subsidizerIdentity = await subsidy.subsidizer.getIdentity();
+  if (!subsidizerIdentity) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'Subsidizer Account does not have an Identity',
+    });
+  }
+  if (!(await subsidizerIdentity.hasValidCdd())) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: 'Subsidizer Account does not have a valid CDD Claim',
+    });
+  }
+}
+
+/** A helper class to ensure a code path is unreachable. For example this can be used for ensuring switch statements are exhaustive */
```

Comment:
```suggestion
/**
 * @hidden
 *
 * Helper class to ensure a code path is unreachable. For example, this can be used for ensuring switch statements are exhaustive 
 */
```